packages/core-plugin-v1/src/runtime.ts

```diff
@@ -1,297 +1,1555 @@
+          // In V1, this would use knowledge.set for each string
+          console.log(`[Compat Layer] Processing non-RAG knowledge`);
+
+          // Filter for just string knowledge
+          const stringKnowledge = this.character.knowledge.filter(
+            (item): item is string => typeof item === 'string'
+          );
+
+          for (const text of stringKnowledge) {
+            try {
+              // Generate a document ID (V1 would use stringToUuid)
+              const id = generateUuidFromString(text.substring(0, 50));
+
+              // Create a V2 document memory (V1 would use this.documentsManager)
+              await this._v2Runtime.createMemory(
+                {
+                  id: id,
+                  entityId: this.agentId as any,
+                  agentId: this.agentId as any,
+                  roomId: this.agentId as any,
+                  content: { text },
+                  metadata: {
+                    type: 'document',
+                    timestamp: Date.now(),
+                  },
+                },
+                'documents'
+              );
+            } catch (error) {
+              console.error(`[Compat Layer] Error processing non-RAG knowledge: ${error.message}`);
+            }
+          }
+        }
+
+        console.log(`[Compat Layer] Knowledge initialization complete`);
+      } catch (error) {
+        console.error(`[Compat Layer] Error during knowledge initialization:`, error);
+      }
+    }
+  }
+
+  getConversationLength(): number {
+    return this._v2Runtime.getConversationLength();
+  }
+
+  getSetting(key: string): string | null {
+    return this._v2Runtime.getSetting(key) as string | null;
   }
 
-  /**
-   * Process the actions of a message.
-   * @param message The message to process.
-   * @param content The content of the message to process actions from.
-   */
   async processActions(
-    message: Memory,
-    responses: Memory[],
+    message: V1Memory,
+    responses: V1Memory[],
     state?: State,
     callback?: HandlerCallback
   ): Promise<void> {
-    return this._runtime.processActions(message, responses, state, callback);
+    throw new Error('processActions not implemented.');
   }
 
-  /**
-   * Evaluate the message and state using the registered evaluators.
-   * @param message The message to evaluate.
-   * @param state The state of the agent.
-   * @param didRespond Whether the agent responded to the message.~
-   * @param callback The handler callback
-   * @returns The results of the evaluation.
-   */
-  async evaluate(message: Memory, state: State, didRespond?: boolean, callback?: HandlerCallback) {
-    // v2 now takes responses: Memory[]
-    return this._runtime.evaluate(message, state, didRespond, callback);
+  async evaluate(
+    message: V1Memory,
+    state?: State,
+    didRespond?: boolean,
+    callback?: HandlerCallback
+  ): Promise<string[] | null> {
+    throw new Error('evaluate not implemented.');
```

Comment:
and here? you removed my implementation without a note, so assuming it was an oversight